cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2534,9 +2533,10 @@ private void writeApplicationSerialized(String namespaceId, String appId, String
     writeApplicationSerialized(namespaceId, appId, versionId, serialized, change, sourceControlMeta, true);
   }
 
-  private void writeApplicationSerialized(String namespaceId, String appId, String versionId,
+  @VisibleForTesting
+  void writeApplicationSerialized(String namespaceId, String appId, String versionId,
```

Comment:
@GnsP After rechecking I think this change is not accurate as sending null in `writeApplicationSerialized` from a caller point of view will be considered as false.

If it is just for testing I think we can write the application normally and then manually update the latest field as null to test. 

The other solution could have been to make latest as nullable in `ApplicationMeta` constructor and resolve it to true if null is passed ( similar to `ChangeDetail`) but that also seems counterintuitive from caller pov